nucypher/blockchain/eth/actors.py

```diff
@@ -260,9 +260,23 @@ def connect_condition_providers(
             providers = set()
             for uri in condition_blockchain_endpoints:
                 provider = self._make_condition_provider(uri)
+                if int(Web3(provider).eth.chain_id) != int(chain_id):
+                    raise self.ActorError(
+                        f"Condition blockchain endpoint {uri} is not on chain {chain_id}"
+                    )
                 providers.add(provider)
             condition_providers[int(chain_id)] = providers
 
+        domain_chain_ids = tuple(chain.id for chain in self.domain.condition_chains)
+        connected_chain_ids = all(
+            _id in condition_providers for _id in domain_chain_ids
+        )
+        if not connected_chain_ids:
+            raise self.ActorError(
+                f"Missing blockchain endpoints for chains: "
+                f"{set(domain_chain_ids) - set(condition_providers)}"
+            )
```

Comment:
Now that we are doing this validation, the `condition_blockchain_endpoints` should no longer be `Optional`. Same for Ursula's `__init__` I guess.

I'm also starting to think `condition_blockchain_endpoints` should be of type `Dict[int, Set[str]]` instead of `Dict[int, List[str]]`. This would filter back to `UrsulaConfiguration` where I'm not sure we are currently properly handling dupes for uris i.e. duplicated uris for the same chain id.

Reply:
Made an update inspired by your comments, which includes some slight refactoring of this function